File: adobe-edge/src/api/details/AdobeErrorDetails.ts
/**
 * Error details information.
 *
 * https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/errordetails.schema.md
 */
export interface AdobeErrorDetails {
  // The error ID.
  name: string;

  // The error source.
  source: ErrorSource;
}

export enum ErrorSource {
  PLAYER = 'player',
  EXTERNAL = 'external',
}

File: adobe-edge/android/src/main/java/com/theoplayer/reactnative/adobe/edge/api/AdobeErrorDetails.kt
package com.theoplayer.reactnative.adobe.edge.api

/**
 * Error details information.
 *
 * @see <a href="https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/errordetails.schema.md">Adobe XDM ErrorDetails Schema</a>
 */
data class AdobeErrorDetails(
  // The error ID.
  val name: String,
  // The error source.
  val source: ErrorSource
)

enum class ErrorSource {
  PLAYER,
  EXTERNAL
}
